Metadata/MetadataConverter.php

```diff
@@ -37,49 +37,56 @@ public function convert(ClassHierarchyMetadata $metadata)
-                );
-            }
+            foreach ($classMetadata->getServices() as $service) {
+                if (null === $previous && null === @$service['parent']) {
+                    $definition = new Definition();
+                } else {
+                    $definition = new DefinitionDecorator(
+                        @$service['parent'] ?: $previous->id
+                    );
+                }
 
-            $definition->setClass($classMetadata->name);
-            if (null !== $classMetadata->scope) {
-                $definition->setScope($classMetadata->scope);
-            }
-            if (null !== $classMetadata->public) {
-                $definition->setPublic($classMetadata->public);
-            }
-            if (null !== $classMetadata->abstract) {
-                $definition->setAbstract($classMetadata->abstract);
-            }
-            if (null !== $classMetadata->arguments) {
-                $definition->setArguments($classMetadata->arguments);
-            }
+                $definition->setClass($classMetadata->name);
+                if (null !== @$service['scope']) {
+                    $definition->setScope(@$service['scope']);
+                }
+                if (null !== @$service['public']) {
+                    $definition->setPublic(@$service['public']);
+                }
+                if (null !== @$service['abstract']) {
+                    $definition->setAbstract(@$service['abstract']);
+                }
+                if (null !== $classMetadata->arguments) {
+                    $definition->setArguments($classMetadata->arguments);
+                }
 
-            $definition->setMethodCalls($classMetadata->methodCalls);
-            $definition->setTags($classMetadata->tags);
-            $definition->setProperties($classMetadata->properties);
+                $definition->setMethodCalls($classMetadata->methodCalls);
+                $definition->setTags($classMetadata->tags);
+                $definition->setProperties($classMetadata->properties);
 
-            if (null === $classMetadata->id) {
-                $classMetadata->id = '_jms_di_extra.unnamed.service_'.$count++;
-            }
+                if (null === @$service['id']) {
+                    $classMetadata->id = '_jms_di_extra.unnamed.service_' . $count++;
+                }
+
+                if ($classMetadata->initMethod) {
+                    if (!method_exists($definition, 'setInitMethod')) {
+                        throw new \RuntimeException(sprintf('@AfterSetup is not available on your Symfony version.'));
+                    }
 
-            if ($classMetadata->initMethod) {
-                if (!method_exists($definition, 'setInitMethod')) {
-                    throw new \RuntimeException(sprintf('@AfterSetup is not available on your Symfony version.'));
+                    $definition->setInitMethod($classMetadata->initMethod);
```

Comment:
no clue where this comes from - this is not present in the symfony repository even in 2011 when it was added

I think it should be removed in another pull request